src/createApiCall.ts

```diff
@@ -89,23 +100,38 @@ export function createApiCall(
 
         const streaming = (currentApiCaller as StreamingApiCaller).descriptor
           ?.streaming;
+
         const retry = thisSettings.retry;
-        if (
-          !streaming &&
-          retry &&
-          retry.retryCodes &&
-          retry.retryCodes.length > 0
-        ) {
-          retry.backoffSettings.initialRpcTimeoutMillis =
-            retry.backoffSettings.initialRpcTimeoutMillis ||
-            thisSettings.timeout;
-          return retryable(
-            func,
-            thisSettings.retry!,
-            thisSettings.otherArgs as GRPCCallOtherArgs,
-            thisSettings.apiName
+        if (!streaming && retry && retry.getResumptionRequestFn) {
+          throw new Error(
+            'Resumption strategy can only be used with server streaming retries'
           );
         }
+        if (!streaming && retry && retry.retryCodesOrShouldRetryFn) {
+          if (
+            retry.retryCodesOrShouldRetryFn instanceof Array &&
```

Comment:
nit: `Array.isArray(retry.retryCodesOrShouldRetryFn)`

Reply:
Guessing that this convention is preferred throughout the code? If so I'll do a find and replace because I know that this is in multiple spots